cai_causal_graph/time_series_causal_graph.py

```diff
@@ -158,6 +158,51 @@ def copy(self, include_meta: bool = True) -> TimeSeriesCausalGraph:
         # cast the graph to TimeSeriesCausalGraph to have the correct metadata
         return TimeSeriesCausalGraph.from_causal_graph(graph)
 
+    def is_stationary(
+        self, return_stationarized_graph: bool = False
```

Comment:
Is stationary is more of an attribute, or method to return a bool, is it wise to conflate that with some returned object? I would assume there would be a separate method "stationarize_graph() -> TimeSeriesCausalGraph" instead.

Reply:
there's a separate `make_stationary` method. The reason why I did like this, it's because when I check if it is stationary I already create the stationary version. So I wanted to condense the computations. But I can split it.

Reply:
I would split it. I don't like this unfortunately and should just return a `bool`. But you could save it as an attribute (that gets reset by your decorator. So if someone calls `is_stationary` and it is `False` and then `make_stationary()` you won't need to generate again.

I would rename to `get_stationary_graph` to match language of `get_summary_graph` and `get_minimal_graph`.

Reply:
I split in two now

Reply:
I appreciate the argument, I've been in this situation many times. But, good coding practices like [SOLID (and single responsibility principle)](https://stackify.com/solid-design-principles/) make life easier downstream. I prefer to use it where possible, even if it mostly refers to classes, having methods change behavior, and also return 1 or 2 objects depending on configuration, creates a bit of complex code. 

Overall, whatever @andrew-causalens wants here.